### ISSUE TITLE
Broken table drag & drop: Tabledrag.js fallback to clientX / clientY not calculating correctly.

### ISSUE THREAD

**Issue by eric2pin** [backdrop/backdrop-issues#5520]
## Description of the bug

If in your theme / jQuery version pageX and pageY aren't defined in tabledrag.js line 632, the fallback to clientX / clientY is not working as expected which results in sudden page scrolling while trying to drag and drop tabledrag items.

## Steps To Reproduce

While performing a table drag (webform configuration in my case, using the webform module, but any tabledrag does the same) with a theme using < jQuery 1.11.1 (1.9.1 in my case), if event.pageX / event.pageY aren't defined on tabledrag.js line 632 (using a Stanford Light base theme converted from Drupal in my case) the fallback to clientX / clientY return the wrong data, the scroll is calculated incorrectly, and the screen / viewport jumps around losing the user's position in their current drag. The fallback for older jQuery versions doesn't appear to work.

## Actual behavior

Viewport is auto-scrolled incorrectly when the user attempts to drag table rows.

## Expected behavior

Auto-scroll is only engaged if a user drags a row near the edges of the viewport.

## Additional information

jQuery 1.9.1
Backdrop CMS 1.21.2
core/misc/tabledrag.js line 632

## Hotfix

For me, adding the following code at ~636 works. Looks like originalEvent.pageX / pageY are defined in jQuery 1.9.1

if (event.originalEvent.pageX || event.originalEvent.pageY) {
  return { x: event.originalEvent.pageX, y: event.originalEvent.pageY };
}

**Comment by indigoxela** [comment]
Hi @eric2pin and welcome to Backdrop.

> ... jQuery 1.9.1...

Currently Backdrop core ships with jQuery v1.12.4. May I ask why your theme ships with an older version? I can't think of any benefit, but am curious enough to ask. :wink:

**Comment by eric2pin** [comment]
Hi @indigoxela thanks for the reply. Our custom theme is a child-theme of the Stanford Light base theme, a child of the Open Framework theme. They have some very particular and outdated dependencies. Both were converted from Drupal 7. This particular theme setup is a client requirement, although at some point those themes could and should be updated to more modern versions.

Updating to a later jQuery version is probably the preferred solution for most -- but the code does say it handles older versions between 1.7 and 1.11, only the fallback doesn't appear to work.

**Comment by indigoxela** [comment]
> This particular theme setup is a client requirement...

I see, you're trapped. (Are you aware, that 1.9.1 has security issues?)

However, it seems like you already have a solution. Mind to file a pull request? Are you familiar with GitHub and pull requests? Please let us know if you need any guidance. It might be tricky to test, tough.

**Comment by eric2pin** [comment]
Hello again @indigoxela , I did attempt to file a pull request here: https://github.com/backdrop/backdrop/pull/3969

Please let me know if I need to take other steps with this / if something about that request doesn't meet standards.

**Comment by indigoxela** [comment]
@eric2pin cool, many thanks.

One thing, that I can't do myself: approve the workflow, so automated tests can run (GitHub insists on that with first contributions). I already notified people via our Zulip chat. Someone will take care of that soon.

Another thing I can't fix - we're over quota with our Tugboat sandboxes. Somebody will take care of that soon, too. These sandboxes come very handy for manual testing.

> Please let me know if I need to take other steps...

Seems like you need a little patience. :wink:

There is one thing you could do: You already linked the Issue in the PR description, which is great. Even better: add the "magical" `Fixes ` directly before that link, then GH adds the PR link to the Issue header automatically.

The code per se looks good, I've left a question, though. Not sure if my JS foo is good enough to review the PR.

**Comment by eric2pin** [comment]
I don't think I know how to approve the workflow. Could you point me to how to do that? I don't think I understand your fixes comment — I'm pretty new to GitHub. Responded to that other comment on the issue.

**Comment by indigoxela** [comment]
> I don't think I know how to approve the workflow.

You can't. I can't either. We have to wait for someone with sufficient permissions. Stay tuned.

**Comment by laryn** [comment]
I approved the workflow run (I think that's just needed for first time submitters).

I also took the liberty of editing the initial comment to include the "Fixes XX" line:

> ![Screen Shot 2022-03-01 at 9 41 39 AM](https://user-images.githubusercontent.com/871421/156200247-e6547efb-0089-48d9-bcda-967c85b96ce2.jpg)

Which then automatically links the issue to the PR in the sidebar:

> ![Screen Shot 2022-03-01 at 9 42 04 AM](https://user-images.githubusercontent.com/871421/156200307-611e8c39-0ad4-4f50-8c0f-d39a59bdeb34.jpg)

**Comment by eric2pin** [comment]
Ah thank you I understand that part now.

**Comment by laryn** [comment]
@eric2pin I have invited you to the Backdrop Documentation team here on Github, just FYI. This gives you some additional permissions in the issue queue.

**Comment by eric2pin** [comment]
Ok great, thanks.

**Comment by klonos** [comment]
@BWPanda has been pinged to cleanup the Tugboat space. Once that's done, we'll need to close the PR, wait 1-2min, then reopen the PR. That should create a PR sandbox ready for testing.

Welcome to the community from me too @eric2pin 👋🏼  ...sorry that your first contribution seems to be hitting some issues on our end. We'll sort these out soon.

**Comment by klonos** [comment]
> The code per se looks good...

Yup, just a small coding standards issue with the inline comment.

**Comment by eric2pin** [comment]
Oops, I didn't consider comment standards. Will double check next time. I found some Contributing to Backdrop docs and will review. Thanks,
